### DISCUSSION TITLE
Question about operations in monitor mode

### DISCUSSION THREAD

**Discussion by amelcon-dot** [abraunegg/onedrive#1946] [Q&A]
Hello, thanks for this wonderful client.
Here I give the log I get running onedrive in monitor mode:
```
2022-May-03 13:25:05.5960788    Starting a sync with OneDrive
2022-May-03 13:25:07.3260899    Uploading differences of v1/amelcon
2022-May-03 13:25:36.239939     Uploading new items of v1/amelcon
2022-May-03 13:26:11.5062946    Sync with OneDrive is complete
2022-May-03 13:36:12.8312521    Uploading differences of v1/amelcon
2022-May-03 13:36:42.2684656    Uploading new items of v1/amelcon
2022-May-03 13:47:19.8244493    Uploading differences of v1/amelcon
2022-May-03 13:47:49.4986231    Uploading new items of v1/amelcon
2022-May-03 13:58:27.8417982    Uploading differences of v1/amelcon
2022-May-03 13:58:57.2112732    Uploading new items of v1/amelcon
2022-May-03 14:09:34.7851231    Uploading differences of v1/amelcon
2022-May-03 14:10:04.1132897    Uploading new items of v1/amelcon
2022-May-03 14:20:40.2141636    Starting a sync with OneDrive
2022-May-03 14:20:41.8699121    Uploading differences of v1/amelcon
2022-May-03 14:21:11.7234311    Uploading new items of v1/amelcon
2022-May-03 14:21:48.1896544    Sync with OneDrive is complete
```
and the same cycle repeats on and on.

I would like to know what the program is doing in all these different phases:
Starting a sync
Uploading differences 
Uploading new items
Sync with OneDrive is complete

It seems it is doing 4 times "Uploading" tasks, before performing a new "Starting a sync".
Could you please explain what are he differences?.
Thanks a lot.

**Comment by abraunegg** [comment]
@amelcon-dot 
What application version is this?

How did you install the application and from what source?

**Reply by amelcon-dot** [reply]
Hello, I installed from OpenSuse Leap:
[OpenSuse Software Onedrive](https://software.opensuse.org/package/onedrive)

using the zypper packed manager.

About the version, it says: 
`onedrive v2.4.16`

Thanks a lot.

**Reply by abraunegg** [reply]
@amelcon-dot 
> onedrive v2.4.16

Firstly, you are running an out-of-date client and v2.4.16 is no longer supported ... please upgrade manually by building a new package yourself or contact the package maintainer to get them to update the package version to v2.4.17 .. or build the client from 'master' yourself.

Secondly .. it is interesting as your log output is not actually representative of what the application is actually doing in that version ... and, you are missing other elements that *should* be written out to the log file or even the systemd logfile .... so curious as to what is going on.

In a *typical* configuration, using application defaults, non-verbose mode, you should get something *similar* to the following:
```
...
2022-May-04 18:29:07.5133251    Starting a sync with OneDrive
2022-May-04 18:29:09.8554101    Sync with OneDrive is complete
2022-May-04 18:30:09.8798586    Starting a sync with OneDrive
2022-May-04 18:30:14.7106614    Sync with OneDrive is complete
2022-May-04 18:31:14.7350326    Starting a sync with OneDrive
2022-May-04 18:31:17.6036948    Sync with OneDrive is complete
2022-May-04 18:32:17.6275854    Starting a sync with OneDrive
2022-May-04 18:32:20.4088409    Sync with OneDrive is complete
2022-May-04 18:33:20.4316305    Starting a sync with OneDrive
2022-May-04 18:33:20.432167     Syncing changes from OneDrive ...
2022-May-04 18:33:25.2963451    Sync with OneDrive is complete
2022-May-04 18:34:25.3200212    Starting a sync with OneDrive
2022-May-04 18:34:29.4065828    Sync with OneDrive is complete
2022-May-04 18:35:29.430873     Starting a sync with OneDrive
2022-May-04 18:35:33.0278777    Sync with OneDrive is complete
2022-May-04 18:36:33.0528543    Starting a sync with OneDrive
2022-May-04 18:36:35.9350944    Sync with OneDrive is complete
2022-May-04 18:37:35.9588114    Starting a sync with OneDrive
2022-May-04 18:37:38.8194305    Sync with OneDrive is complete
2022-May-04 18:38:38.843336     Starting a sync with OneDrive
2022-May-04 18:38:38.8440114    Syncing changes from OneDrive ...
2022-May-04 18:38:41.6884213    Sync with OneDrive is complete
2022-May-04 18:39:41.7153249    Starting a sync with OneDrive
2022-May-04 18:39:44.5074365    Sync with OneDrive is complete
2022-May-04 18:40:44.532944     Starting a sync with OneDrive
2022-May-04 18:40:48.0974593    Sync with OneDrive is complete
2022-May-04 18:41:48.1243778    Starting a sync with OneDrive
2022-May-04 18:41:52.400069     Sync with OneDrive is complete
2022-May-04 18:42:52.4237449    Starting a sync with OneDrive
2022-May-04 18:42:57.0181306    Sync with OneDrive is complete
2022-May-04 18:43:57.0421663    Starting a sync with OneDrive
2022-May-04 18:43:57.0427139    Syncing changes from OneDrive ...
2022-May-04 18:43:59.8623681    Sync with OneDrive is complete
2022-May-04 18:44:59.8873294    Starting a sync with OneDrive
2022-May-04 18:45:03.0030058    Sync with OneDrive is complete
2022-May-04 18:46:03.0290458    Starting a sync with OneDrive
2022-May-04 18:46:06.3836948    Sync with OneDrive is complete
2022-May-04 18:47:06.4072899    Starting a sync with OneDrive
2022-May-04 18:47:09.9866796    Sync with OneDrive is complete
2022-May-04 18:48:10.0112659    Starting a sync with OneDrive
2022-May-04 18:48:13.160902     Sync with OneDrive is complete
2022-May-04 18:49:13.1875358    Starting a sync with OneDrive
2022-May-04 18:49:13.1889873    Syncing changes from OneDrive ...
2022-May-04 18:49:17.2253538    Sync with OneDrive is complete
...
```

**Note:** my  configuration is that I am syncing every 60 seconds, to illustrate what it is meant to be doing ... or at least what should be written to the log / console output.

When using `--verbose` .. this increases to the following:
```
...
2022-May-04 19:01:16.5590501    Updated Remaining Free Space: 5364084328
2022-May-04 19:01:17.6192907    Performing Database Consistency Integrity Check .. 
2022-May-04 19:01:17.6197506    Uploading differences of ~/OneDrivePersonal
2022-May-04 19:01:17.6206328    Processing .
2022-May-04 19:01:17.6209302    The directory has not changed
...
2022-May-04 19:01:17.6511138    The file has not changed
2022-May-04 19:01:17.6518003    Processing random_images/rxBrpjYZJOHoe5C87yCYUq4QekeOaOmE/image8.png
2022-May-04 19:01:17.6521691    The file has not changed
2022-May-04 19:01:17.65275      Processing Documents
2022-May-04 19:01:17.6530582    The directory has not changed
...
2022-May-04 19:01:17.6615504    Processing Documents/Music/audio001.mp3
2022-May-04 19:01:17.6619095    The file has not changed
2022-May-04 19:01:17.6624986    Processing Music
2022-May-04 19:01:17.6628023    The directory has not changed
2022-May-04 19:01:17.6637113    Processing Music/audio003.mp3
2022-May-04 19:01:17.6640725    The file has not changed
2022-May-04 19:01:17.6646216    Processing Images
2022-May-04 19:01:17.6649179    The directory has not changed
2022-May-04 19:01:17.6656643    Processing Images/image003.jpg
2022-May-04 19:01:17.666004     The file has not changed
2022-May-04 19:01:17.6664897    Uploading new items of ~/OneDrivePersonal
2022-May-04 19:01:17.6875453    Skipping item - excluded by skip_file config: ./file1.tmp
2022-May-04 19:01:17.6878788    Applying changes of Path ID: 66D53BE8A5056ECA!101
2022-May-04 19:01:18.2819935    Updated Remaining Free Space: 5364084328
2022-May-04 19:01:22.3677114    Processing 1 OneDrive items to ensure consistent local state due to a full scan being requested
2022-May-04 19:01:22.3712558    Sync with OneDrive is complete
2022-May-04 19:02:22.4016473    Starting a sync with OneDrive
2022-May-04 19:02:22.402913     Applying changes of Path ID: 66D53BE8A5056ECA!101
2022-May-04 19:02:22.8015062    Updated Remaining Free Space: 5364084328
2022-May-04 19:02:24.1266457    Uploading new items of ~/OneDrivePersonal
2022-May-04 19:02:24.1487906    Skipping item - excluded by skip_file config: ./file1.tmp
2022-May-04 19:02:24.1490871    Applying changes of Path ID: 66D53BE8A5056ECA!101
2022-May-04 19:02:24.7607973    Updated Remaining Free Space: 5364084328
2022-May-04 19:02:25.6414522    Sync with OneDrive is complete
...
```

So to answer this:
> Starting a sync

This is indicating that the client is starting a sync with OneDrive 


> Uploading differences

This performs the local database consistency check - ensuring that the items on disk are, what they say they are, and valid from a date/timestamp & hash perspective - ie  has something changed? If it has, upload it or delete it or perform the appropriate action

> Uploading new items

Seek the local filesystem for new files

> Sync with OneDrive is complete

This is indicating that the sync is complete


Can you please post your 'config' file and a full `--verbose` log file please.

**Comment by amelcon-dot** [comment]
OK, thanks for your explanations. Yes, I wonder if OpenSuse has changed anything. I will try to run the last version.
I could not produce a clean --verbose log file; I think is because I have many GB (over 200 GB) in onedrive and it needs to scan them all. I tried to use --single-directory to set up a small directory test with a couple of files to sync, but it still gives tons of: Remote change discarded - not in --single-directory sync scope (not in DB). 

As for the config file, I think I did not do anything special. Here it goes:
```
# sync_dir = "~/OneDrive"
sync_dir = "~"
# skip_file = "~*|.~*|*.tmp"
skip_file = "~*|.~*|*.msf|*.tmp|*.aux|*.bbl|*.blg|*.dvi|*.log|*.soc|Thumbs.db*"
monitor_interval = "600"
# skip_dir = ""
skip_dir = "*.aedtresults|*_lib"
# log_dir = "/var/log/onedrive/"
log_dir = "~/logs/onedriveAMELCON/"
# drive_id = ""
# upload_only = "false"
# check_nomount = "false"
check_nosync = "true"
# download_only = "false"
# disable_notifications = "false"
# disable_upload_validation = "false"
# enable_logging = "false"
# force_http_11 = "false"
# force_http_2 = "false"
# local_first = "false"
# no_remote_delete = "false"
skip_symlinks = "true"
# debug_https = "false"
skip_dotfiles = "true"
# dry_run = "false"
# min_notify_changes = "5"
# monitor_log_frequency = "5"
# monitor_fullscan_frequency = "10"
# sync_root_files = "false"
# classify_as_big_delete = "1000"
# user_agent = ""
# remove_source_files = "false"
# skip_dir_strict_match = "false"
# application_id = ""
# resync = "false"
# bypass_data_preservation = "false"
# azure_ad_endpoint = ""
# azure_tenant_id = "common"
# sync_business_shared_folders = "false"
# sync_dir_permissions = "700"
# sync_file_permissions = "600"
# rate_limit = "131072"
``` 

I run the client with this command:
`
onedrive --monitor 
         --confdir="~/.config/onedriveAMELCON" --enable-logging 
         --single-directory "v1/amelcon"
`

**Comment by abraunegg** [comment]
@amelcon-dot 
Please provide the log file as requested ... I suspect you potentially have installed an earlier version of the client and it is not v2.4.16 ... the OpenSuSE Leap version of the client are:

* OpenSuSE Leap 15.2 = onedrive v2.4.3
* OpenSuSE Leap 15.3 = onedrive v2.4.8
* OpenSuSE Leap 15.4 = onedrive v2.4.14
* OpenSuSE Tumbleweed = onedrive v2.4.16

Given your config file includes `force_http_11` which was removed in v2.4.11 ... I suspect you are running OpenSuSE Leap 15.3 thus potentially running client version v2.4.8 ....

Please also provide the output of the following commands in addition to a log file as requested:
* `onedrive --version`
* `onedrive --display-config`

**Comment by amelcon-dot** [comment]
Hello,

The config file comes from an older version; it is true that when I update the client I did not change the config file. So this config file indeed comes from an older version; I did not think about changing the config file when updating, sorry.

Here are the output you requested:
`onedrive --version`
```
onedrive v2.4.16
```
`onedrive --display-config --confdir="~/.config/onedriveAMELCON"`
```
Configuration file successfully loaded
onedrive version                       = v2.4.16
Config path                            = /home/amelcon/.config/onedriveAMELCON
Config file found in config path       = true
Config option 'check_nosync'           = true
Config option 'sync_dir'               = /home/amelcon
Config option 'skip_dir'               = *.aedtresults|*_lib
Config option 'skip_file'              = ~*|.~*|*.msf|*.tmp|*.aux|*.bbl|*.blg|*.dvi|*.log|*.soc|Thumbs.db*
Config option 'skip_dotfiles'          = true
Config option 'skip_symlinks'          = true
Config option 'monitor_interval'       = 600
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /home/amelcon/logs/onedriveAMELCON/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = false
Business Shared Folders configured     = false
```
I also attach the log file as requested.

Thanks a lot.
[log_file.zip](https://github.com/abraunegg/onedrive/files/8629302/log_file.zip)

**Reply by amelcon-dot** [reply]
In any case, I think the best is that I try to upgrade to 2.4.17, and also change the config file to the last version. I think by doing this I will come to normal operation. Sorry for the issue and thanks a lot for the support.

**Reply by amelcon-dot** [reply]
A part from this issue detected by you, after testing a coupled of days I can say that the program seems to work very well in monitor mode.
It performs a synchronization every 10 min as specified in the config file (monitor_interval = "600").
In between synchronizations it waits, but somehow it picks any local change and uploads the change almost instantly (this is very remarkable; cannot image how is this possible).
In the next synchronization it picks any change taking place in remote, and downloads the change to the local disk.

Overall I find the operation amazing, congratulations, and thanks for your help.
